docs/sources/flow/reference/components/pyroscope.ebpf.md

```diff
@@ -0,0 +1,178 @@
+| `container_id_cache_size` | `int`                    | The size of the pid -> container ID  LRU cache                         | 64      | no       |       
+| `collect_user_profile`    | `bool`                   | A flag to enable/disable collection of userspace profiles              | true    | no       |       
+| `collect_kernel_profile`  | `bool`                   | A flag to enable/disable collection of kernelspace profiles            | true    | no       |       
+
+## Exported fields
+
+`pyroscope.ebpf` does not export any fields that can be referenced by other
+components.
+
+## Component health
+
+`pyroscope.ebpf` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`pyroscope.ebpf` does not expose any component-specific debug information.
+
+## Debug metrics
+
+* `pyroscope_fanout_latency` (histogram): Write latency for sending to direct and indirect components.
+* `pyroscope_ebpf_pid_cache_hit_total` (counter): Total number of ebpf symbolizer pid cache hit.
+* `pyroscope_ebpf_pid_cache_miss_total` (counter): Total number of ebpf symbolizer pid cache miss.
+* `pyroscope_ebpf_elf_cache_build_id_hit_total` (counter): Total number of ebpf symbolizer elf cache (build-id) hit.
+* `pyroscope_ebpf_elf_cache_build_id_miss_total` (counter): Total number of ebpf symbolizer elf cache (build-id) miss.
+* `pyroscope_ebpf_elf_cache_stat_hit_total` (counter): Total number of ebpf symbolizer elf cache (stat) hit.
+* `pyroscope_ebpf_elf_cache_stat_miss_total` (counter): Total number of ebpf symbolizer elf cache (stat) miss.
+* `pyroscope_ebpf_container_id_cache_hit_total` (counter): Total number of ebpf target finder container id cache hit.
+* `pyroscope_ebpf_container_id_cache_miss_total` (counter): Total number of ebpf target finder container id cache miss.
+
+## Profile collecting behavior
+
+The `pyroscope.ebpf` component is designed to collect stacktraces associated with a process running on the current host.
+Stack traces are collected according to the defined `sample_rate`, meaning traces are gathered this many times
+per second.
+
+The following labels are automatically injected to the collected profiles if not found and can help pin down a
+profiling target.
+
+| Label              | Description                                                                                                                                                                                            |
+|--------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `service_name`     | Pyroscope service name. It's automatically selected from discovery meta labels if possible. Otherwise defaults to the fully formed component name or a value from `pyroscope.ebpf.service_name` field. |
```

Comment:
is `pyroscope.ebpf.service_name` a label name ? Why not keep the `service_name` if it's provided via the target labels ?

Reply:
no, it was a leftover. 

Now we have the following section

```
### Service name

The special label `service_name` is required and must always be present. If it's not specified, it is
attempted to be inferred from multiple sources:

- `__meta_kubernetes_pod_annotation_pyroscope_io_service_name` which is a `pyroscope.io/service_name` pod annotation.
- `__meta_kubernetes_namespace` and `__meta_kubernetes_pod_container_name`
- `__meta_docker_container_name`

If `service_name` is not specified and could not be inferred it is set to `unspecified`.
```

And the following row in the table

```
| `service_name`     | Pyroscope service name. It's automatically selected from discovery meta labels if possible. Otherwise defaults to `unspecified`. |
```

Does it look good now?